### PR TITLE
Merge Norm into LinearAlgebra and deprecate Norm

### PR DESCRIPTION
**Pull request by ben-albrecht**
Merge Norm into LinearAlgebra and deprecate Norm module.

Some slight interface modifications have been made to the Norm module:

- `p: normType` argument is now a param
  - This is required to ensure LAPACK will only be required when using norm-2 with 2D arrays when we add this implementation.
- Added a `default` value to the `normType` enum
  - This eliminates the need for multiple overloads in the interface and better follows patterns in other linear algebra libraries.

Other changes include:

- Norm module has been properly deprecated (warning in documentation, compiler warning, deprecation test added).
- All tests using Norm now use LinearAlgebra instead.

Future work includes:

- Consolidating Norm tests and migrating them to the LinearAlgebra testing directory
- Implementing the 2-norm for 2D arrays - I have added a commented out implementation, which I would like to pursue as a follow-up effort.

**Testing**

- [x] linux64 testing

### CODE REVIEW

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -1194,16 +1193,16 @@ private proc _lu (in A: [?Adom] ?eltType) {
...
+
+  `L` and `U` are stored in the same matrix `LU` where
   the unit diagonal elements of L are not stored.
-  
-  `ipiv` contains the pivot indices such that row i of `A` 
+
+  `ipiv` contains the pivot indices such that row i of `A`
   was interchanged with row `ipiv(i)`.
-  
+
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl:1205]: Maybe this one's not needed? Clearly it's personal taste, and you can ignore :)

File: modules/packages/LinearAlgebra.chpl [hunk 2]
```diff
@@ -1270,54 +1269,135 @@ proc det (A: [?Adom] ?eltType) {
...
+  :rtype: x.eltType
+ */
+proc _norm(x: [?D], param p: normType) where x.rank == 2 {
+  select (p) {
+  when normType.norm1 do
+    return max reduce forall j in D.dim(1) do (+ reduce abs(x[D.dim(0), j..j]));
+
+  when normType.norm2 {
+    halt("2-norm for 2D arrays are not yet implemented");
```
Comment (e-kayrakli) [inline modules/packages/LinearAlgebra.chpl:1344]: Can't we make this a compile-time error?

Comment (ben-albrecht) [reply modules/packages/LinearAlgebra.chpl:1344]: Ah yes, good point!

File: modules/packages/LinearAlgebra.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/packages/LinearAlgebra.chpl:1205]: I only deleted whitespace, but I'll remove this newline for you :)

Comment (e-kayrakli) [reply modules/packages/LinearAlgebra.chpl:1205]: You are very kind :)

**Pull request merged**